go/internal/cmd.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package internal
+
+// #include "../lib.h"
+import "C"
+
+import (
+	"fmt"
+
+	"github.com/valkey-io/valkey-glide/go/v2/config"
+	"github.com/valkey-io/valkey-glide/go/v2/internal/errors"
+)
+
+type Batch struct {
+	Commands []Cmd
+	IsAtomic bool
+	Errors   []string // errors processing command args, spotted while batch is filled
+}
+
+type Cmd struct {
+	RequestType uint32 // TODO why C.RequestType doesn't work?
```

Comment:
C.RequestType won't work if you are using it across packages. When CGO types are imported, they become a type specific to that package. The same type imported in two different packages will be recognized in go as two different incompatible types. The appropriate solution is what I started last week to pull our common types into a single FFI go wrapper package which could be used across packages without issue. We will need to address that in 2.1 as it is very large change.